app/heron/factories/plate.rb

```diff
@@ -6,70 +6,62 @@ module Factories
+      include Concerns::RecipientsCoordinates
+      include Concerns::Contents
 
-      attr_accessor :plate
+      attr_accessor :plate, :plate_purpose
 
-      validate :plate_purpose_xor_plate_purpose_uuid
-      validate :plate_purpose_set
-      validate :plate_purpose_uuid_set
-
-      validate :validate_wells_content
+      validate :plate_purpose_exists
 
       def initialize(params)
         @params = params
       end
 
-      def barcode
-        @params[:barcode]
+      def recipients_key
+        :wells
       end
 
-      def wells_content
-        @wells_content ||= ::Heron::Factories::WellsContent.new(@params[:wells_content], @params[:study_uuid])
+      def content_factory
+        ::Heron::Factories::Sample
       end
 
-      def validate_wells_content
-        return if wells_content.valid?
-
-        errors.add(:wells_content, wells_content.errors.full_messages)
+      def barcode
+        @params[:barcode]
       end
 
-      def plate_purpose_xor_plate_purpose_uuid
-        return if @params.key?(:plate_purpose_uuid) ^ @params.key?(:plate_purpose)
+      def save
+        return false unless valid?
+
+        ActiveRecord::Base.transaction do
+          @plate = plate_purpose.create!
+
+          Barcode.create!(asset: @plate, barcode: barcode, format: barcode_format)
 
-        if @params.key?(:plate_purpose_uuid) && @params.key?(:plate_purpose)
-          errors.add(:base, 'You cannot define plate_purpose_uuid and plate_purpose at same time')
-        else
-          errors.add(:base, 'You have to define either plate_purpose_uuid or plate_purpose')
+          create_contents!
         end
+        true
       end
 
-      def plate_purpose_set
-        return unless @params.key?(:plate_purpose)
+      private
 
-        @plate_purpose ||= @params[:plate_purpose]
-        errors.add(:base, "Plate purpose can't be blank") unless @plate_purpose
+      def create_contents!
+        add_aliquots_into_locations(containers_for_locations)
       end
 
-      def plate_purpose_uuid_set
-        return unless @params.key?(:plate_purpose_uuid)
-
+      def plate_purpose_exists
+        unless @params.key?(:plate_purpose_uuid)
+          errors.add(:base, 'Plate purpose uuid not defined')
+          return
+        end
         @plate_purpose ||= PlatePurpose.with_uuid(@params[:plate_purpose_uuid]).first
         errors.add(:base, "Plate purpose for uuid (#{@params[:plate_purpose_uuid]}) do not exist") unless @plate_purpose
       end
 
-      attr_reader :plate_purpose
-
-      def save
-        return false unless valid?
-
-        ActiveRecord::Base.transaction do
-          @plate = plate_purpose.create!
-
-          Barcode.create!(asset: @plate, barcode: barcode, format: barcode_format)
-
-          wells_content.add_aliquots_into_plate(plate)
+      def containers_for_locations
+        @plate.wells.each_with_object({}) do |well, memo|
+          memo[unpad_coordinate(well.map.description)] = well
```

Comment:
Could you use Map.strip_description instead?

Reply:
Ok, added in use under unpad_coordinate